src/components/Display/CharTable.js

```diff
@@ -0,0 +1,91 @@
+import {
+  Table,
+  TableContainer,
+  TableHead,
+  TableBody,
+  TableRow,
+  TableCell,
+  Paper,
+  Avatar,
+  Box,
+} from '@mui/material';
+import React, { useState } from 'react';
+import CharModal from '../Modal/CharModal';
+
+const CharTable = ({ data }) => {
+  const [open, setOpen] = useState(false);
+  const [modalInfo, setModalInfo] = useState(null);
+
+  const handleOpen = row => {
+    setOpen(true);
+    setModalInfo(row);
```

Comment:
Just a general best practice, use `batch` to make them execute in the same `render` run
```suggestion
    batch(() => {
        setOpen(true);
        setModalInfo(row);
    })
```